packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -773,18 +798,41 @@ void _parseIssueInStdout(XcodeBuildExecution xcodeBuildExecution, Logger logger,
       && (result.stdout?.contains('requires a provisioning profile. Select a provisioning profile in the Signing & Capabilities editor') ?? false)) {
     logger.printError(noProvisioningProfileInstruction, emphasis: true);
   }
+
+  if (stderr != null && stderr.contains('Ineligible destinations')) {
+    final String? version = _parseMissingPlatform(stderr);
+      if (version != null) {
+        logger.printError(missingPlatformInstructions(version), emphasis: true);
+      }
+  }
+}
+
+String? _parseMissingPlatform(String message) {
+  final RegExp pattern = RegExp(r'error:(.*?) is not installed\. To use with Xcode, first download and install the platform');
```

Comment:
this seems a bit fragile (e.g. apple updates this string). Is there any error code you can use instead?

Reply:
Unfortunately no, there's nothing identifiable about it other than the message

Here's examples of the output/xcresult:

https://github.com/flutter/flutter/pull/130286/files#diff-d3f6153fa6dc8bb8da44d3b36e987346a3f92af86eed53babfcd91088f7e0d07R256-R270
https://github.com/flutter/flutter/pull/130286/files#diff-6dbb589bcbf47676289bc9a0c2103a8eef5debc5ee3090f6ab750c8af13a57b6R433